Added health check

RGanor opened the pull request:
Add health check API that also check the authoritative api.

jbe-dw commented on powerdnsadmin/routes/api.py:
```diff
@@ -1182,3 +1182,37 @@ def sync_domains():
+    if resp.status_code == 200:
+        return make_response("good", 200)
+    else:
+        return make_response("bad", 503)
```
I'm not sure if we should use 503 here, it's not PDA failing but the backend

jbe-dw reviewed (commented):
I took time to give my opinion.

Regards

jbe-dw commented on powerdnsadmin/routes/api.py:
```diff
@@ -1182,3 +1182,37 @@ def sync_domains():
+        current_app.logger.error("No domain found to query a health check")
+        raise (HealthCheckFail)
+
+    pdns_api_url = Setting().get('pdns_api_url')
+    pdns_api_key = Setting().get('pdns_api_key')
+    pdns_version = Setting().get('pdns_version')
+    api_uri_with_prefix = utils.pdns_api_extended_uri(pdns_version)
+    api_uri = '/servers/localhost/zones/{}'.format(domain_to_query.name)
+    headers = {}
+    headers['X-API-Key'] = pdns_api_key
+
+    try:
+        resp = utils.fetch_remote(urljoin(pdns_api_url, api_uri_with_prefix + api_uri),
+                                method='GET',
+                                headers=headers,
+                                accept='application/json; q=1',
+                                verify=Setting().get('verify_ssl_connections'))
+                                
+    except Exception as e:
+        current_app.logger.error("Health Check - Failed to query authoritative server for domain {}".format(domain_to_query.name))
+        return make_response("bad", 503)
```
This piece of code is already implemented in get_domain_info(). Did you try the code I posted ?

RGanor commented:
@jbe-dw I have updated the code to use the domain model function and added authentication requirement.
Now, after we removed the `/ping` authentication, I think it make sense for this function to be more robust in its check, but because it is querying the pdns auth api (which can be resource intensive some time), the authentication is required to protect this endpoint.
I think that `/ping` and `/api/v1/health` should co-exist, each with different propose.

jbe-dw reviewed (changes requested):
As we discussed, i'm waiting for an alternative message if there is no domain to query

RGanor commented:
Hey @jbe-dw , thanks for bumping this up. I will be able fix this soon.
